node/core/approval-voting/src/criteria.rs

```diff
@@ -641,72 +674,81 @@ pub(crate) fn check_assignment_cert(
 
 	let &(ref vrf_output, ref vrf_proof) = &assignment.vrf;
 	match &assignment.kind {
-		AssignmentCertKindV2::RelayVRFModuloCompact { sample, core_indices } => {
-			if *sample >= config.relay_vrf_modulo_samples {
-				return Err(InvalidAssignment(Reason::SampleOutOfBounds))
-			}
-
+		AssignmentCertKindV2::RelayVRFModuloCompact => {
 			let (vrf_in_out, _) = public
-				.vrf_verify_extra(
-					relay_vrf_modulo_transcript(relay_vrf_story, *sample),
+				.vrf_verify(
+					relay_vrf_modulo_transcript_v2(relay_vrf_story),
 					&vrf_output.0,
 					&vrf_proof.0,
-					assigned_cores_transcript(core_indices),
 				)
 				.map_err(|_| InvalidAssignment(Reason::VRFModuloOutputMismatch))?;
 
-			let resulting_cores = relay_vrf_modulo_cores(&vrf_in_out, *sample + 1, config.n_cores);
-
-			// TODO: Enforce that all claimable cores are claimed, or include refused cores.
-			// Currently validators can opt out of checking specific cores.
-			// This is the same issue to how validator can opt out and not send their assignments in the first place.
+			// Get unique core assignments from the VRF wrt `config.n_cores`.
+			let vrf_unique_cores = relay_vrf_modulo_cores(
+				&vrf_in_out,
+				config.relay_vrf_modulo_samples,
+				config.n_cores,
+			);
 
-			// Ensure that the `vrf_in_out` actually includes all of the claimed cores.
-			if claimed_core_indices
+			// Filter out cores in which the validator is in the backing group.
```

Comment:
why do we need this check now and didn't have this before? honest backing validators should not be participating in approvals

Reply:
nvm, we were checking this before

Reply:
Yeah, there is also a secondary check for claimed candidate indices because this only checks the crypto part, and can yield cores that did not became free in that block(a candidate was included).

Reply:
I'll add a comment